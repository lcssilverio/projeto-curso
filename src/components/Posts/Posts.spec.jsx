import { Posts } from ".";
import { render, screen } from "@testing-library/react";

const mockPosts = {
  posts: [
    {
      id: 1,
      title: "Title 1",
      body: 'Body 1',
      covert: 'img/img1.png'
    },
    {
      id: 2,
      title: "Title 2",
      body: 'Body 2',
      covert: 'img/img2.png'
    },
    {
      id: 3,
      title: "Title 3",
      body: 'Body 3',
      covert: 'img/img3.png'
    },
    {
      id: 4,
      title: "Title 4",
      body: 'Body 4',
      covert: 'img/img4.png'
    }
  ]
}

describe("Tests to Posts component", () => {
  test("should render Posts components", () => {
    render(<Posts {...mockPosts}/>);

    expect(screen.getAllByRole('heading', {name:  /Title/i})).toHaveLength(4)
    expect(screen.getAllByRole('img', {name:  /Title/i})).toHaveLength(4)
    expect(screen.getAllByText(/body/i)).toHaveLength(4)
  });

  test("should not render Posts", () => {
    render(<Posts />);

    expect(screen.queryByRole('heading', {name:  /Title/i})).not.toBeInTheDocument()
  });

  test("should match snapshot", () => {
    const {container} = render(<Posts {...mockPosts}/>);
    expect(container).toMatchSnapshot()
  });
});
