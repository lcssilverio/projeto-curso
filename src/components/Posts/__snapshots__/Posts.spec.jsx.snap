// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests to Posts component should match snapshot 1`] = `
<div
  class="posts"
>
  <div
    class="post"
  >
    <img
      alt="Title 1"
    />
    <div
      class="post-content"
    >
      <h2>
        Title 1
         
        1
      </h2>
      <p>
        Body 1
      </p>
    </div>
  </div>
  <div
    class="post"
  >
    <img
      alt="Title 2"
    />
    <div
      class="post-content"
    >
      <h2>
        Title 2
         
        2
      </h2>
      <p>
        Body 2
      </p>
    </div>
  </div>
  <div
    class="post"
  >
    <img
      alt="Title 3"
    />
    <div
      class="post-content"
    >
      <h2>
        Title 3
         
        3
      </h2>
      <p>
        Body 3
      </p>
    </div>
  </div>
  <div
    class="post"
  >
    <img
      alt="Title 4"
    />
    <div
      class="post-content"
    >
      <h2>
        Title 4
         
        4
      </h2>
      <p>
        Body 4
      </p>
    </div>
  </div>
</div>
`;
